client/driver/executor/test_harness_test.go

```diff
@@ -14,6 +15,57 @@ import (
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
+func TestMain(m *testing.M) {
+	switch os.Getenv("TEST_MAIN") {
+	case "app":
+		appMain()
+	default:
+		os.Exit(m.Run())
+	}
+}
+
+func appMain() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "no command provided")
+		os.Exit(1)
+	}
+	args := os.Args[1:]
+	shift := func() string {
+		s := args[0]
+		args = args[1:]
+		return s
+	}
+	for len(args) > 0 {
+		switch cmd := shift(); cmd {
+		case "sleep":
+			if len(args) < 1 {
```

Comment:
I would prefer if you put a comment on each case describing the behavior and how many arguments they take etc. It is essentially a function and we should document it as such.